### PR TITLE
chore: Quality of Life: xtask all improvements.

### PR DESCRIPTION
**Pull request by o0Ignition0o**
`cargo xtask all` will now actually format your code and apply the license update for you!

### CODE REVIEW

**Review by garypen — CHANGES_REQUESTED** [review]
I'm not sure if my comments are correct, but I'm asking anyway.

File: xtask/src/commands/lint.rs [hunk 1]
```diff
@@ -7,9 +7,33 @@ pub struct Lint {}
...
+            cargo!(["fmt", "--all"]);
+            eprintln!(
+                "🧹 cargo fmt job is complete 🧹\n\
+                Commit the changes and you should be good to go!"
+            );
+        };
+
+        cargo!(["clippy", "--all", "--tests", "--", "-D", "warnings"]);
+
+        Ok(())
```
[the remark this answers is not in the document]
Comment (o0Ignition0o) [reply xtask/src/commands/lint.rs:27]: Totally agree it doesn't read well, let's see if i can do better.

Basically what i m trying to convey is "run cargo fmt --check. if it fails, notify the developper they will have a diff and they shall commit it, and run the actual cargo fmt"

File: xtask/src/commands/all.rs [hunk 2]
```diff
@@ -15,11 +15,11 @@ pub struct All {
...
+        eprintln!("Running tests...");
+        self.test.run()?;
         eprintln!("Checking format and clippy...");
-        self.lint.run()?;
+        self.lint.run_local()?;
         eprintln!("Checking licenses...");
-        self.compliance.run()?;
-        eprintln!("Running tests...");
-        self.test.run()
+        self.compliance.run_local()
```
[the remark this answers is not in the document]
Comment (o0Ignition0o) [reply xtask/src/commands/all.rs:23]: if someone checks in badly formatted code, the ci will run `cargo xtask lint` (and not `cargo xtask all`) which will run `cargo fmt --check` and fail the CI

[the remark this answers is not in the document]
Comment (Geal) [reply xtask/src/commands/all.rs:23]: yup, it helps locally but the CI guarantees everything

**Review by Geal — APPROVED** [review]

**Pull request merged**